examples/docker/docker-compose.yaml

```diff
@@ -82,11 +74,14 @@ services:
       - "8080:8080"
 
   jupyterhub:
-    image: quansight/conda-store
+    build:
+      context: dockerfiles
+      dockerfile: jupyterhub.Dockerfile
+      args:
+        python_version: 3.11
```

Comment:
Not sure if this is actually something we need to do here (or if there's some reason to be using 3.11), but I vote to target the latest stable version if possible (3.13)

Reply:
I get a conda error when running python 3.13
```
> [jupyterhub 3/7] RUN mamba create --name conda-store     python=3.13 nb_conda_kernels nodejs=18 yarn constructor conda-store     --channel conda-forge -y &&     mamba clean --all -y &&     conda clean --force-pkgs-dirs:                     
14.27                                                                            
14.27 Looking for: ['python=3.13', 'nb_conda_kernels', 'nodejs=18', 'yarn', 'constructor', 'conda-store']
14.27 
14.27 Could not solve for environment specs
14.27 The following packages are incompatible
14.27 ├─ constructor is installable with the potential options
14.27 │  ├─ constructor [1.3.0|1.3.1|...|3.0.1] would require
14.27 │  │  └─ python [2.7* |>=2.7,<2.8.0a0 ], which can be installed;
14.27 │  ├─ constructor [1.3.0|1.3.1|1.3.2|1.3.3|1.3.4] would require
14.27 │  │  └─ python 3.4* , which can be installed;
14.27 │  ├─ constructor [1.3.0|1.3.1|...|2.2.0] would require
14.27 │  │  └─ python [3.5* |>=3.5,<3.6.0a0 ], which can be installed;
14.27 │  ├─ constructor [1.5.5|2.0.0|2.0.1|2.0.2|2.0.3] would require
14.27 │  │  └─ python 3.6* , which can be installed;
14.27 │  ├─ constructor [2.1.0|2.2.0|...|3.2.1] would require
14.27 │  │  └─ python >=3.6,<3.7.0a0 , which can be installed;
14.27 │  ├─ constructor [2.2.0|2.3.0|...|3.3.1] would require
14.27 │  │  └─ python >=3.7,<3.8.0a0 , which can be installed;
14.27 │  ├─ constructor [3.0.2|3.1.0|...|3.3.1] would require
14.27 │  │  └─ python >=3.8,<3.9.0a0 , which can be installed;
14.27 │  ├─ constructor [3.1.0|3.2.0|3.2.1] would require
14.27 │  │  └─ python_abi 3.6 *_pypy36_pp73, which requires
14.27 │  │     └─ python 3.6.* *_73_pypy, which can be installed;
14.27 │  ├─ constructor [3.2.0|3.2.1|3.2.2|3.3.1] would require
14.27 │  │  └─ python >=3.9,<3.10.0a0 , which can be installed;
14.27 │  ├─ constructor [3.2.2|3.3.1] would require
14.27 │  │  └─ python >=3.10,<3.11.0a0 , which can be installed;
14.27 │  ├─ constructor [3.10.0|3.11.0|...|3.9.3] would require
14.27 │  │  └─ __win, which is missing on the system;
14.27 │  ├─ constructor [3.3.1|3.4.1|3.4.2|3.4.3] would require
14.27 │  │  └─ conda [>=4.6,<23.1.0 |>=4.6,<23.1.0a0 ] with the potential options
14.27 │  │     ├─ conda [22.11.1|22.9.0|...|4.14.0] would require
14.27 │  │     │  └─ python >=3.10,<3.11.0a0 , which can be installed;
14.27 │  │     ├─ conda [22.11.1|22.9.0|...|24.9.2] would require
14.27 │  │     │  └─ python >=3.11,<3.12.0a0 , which can be installed;
14.27 │  │     ├─ conda [22.9.0|4.10.0|...|4.9.2] would require
14.27 │  │     │  └─ python >=3.7,<3.8.0a0 , which can be installed;
14.27 │  │     ├─ conda [22.11.1|22.9.0|4.14.0] would require
14.27 │  │     │  └─ python_abi 3.8 *_pypy38_pp73, which requires
14.27 │  │     │     └─ python 3.8.* *_73_pypy, which can be installed;
14.27 │  │     ├─ conda [22.11.1|22.9.0|...|4.9.2] would require
14.27 │  │     │  └─ python >=3.8,<3.9.0a0 , which can be installed;
14.27 │  │     ├─ conda [22.11.1|22.9.0|...|4.14.0] would require
14.27 │  │     │  ├─ pypy3.9 [>=7.3.13 |>=7.3.15 ], which requires
14.27 │  │     │  │  └─ python 3.9.* *_73_pypy, which can be installed;
14.27 │  │     │  └─ python_abi 3.9 *_pypy39_pp73, which requires
14.27 │  │     │     └─ python 3.9.* *_73_pypy, which can be installed;
14.27 │  │     ├─ conda [22.11.1|22.9.0|...|4.9.2] would require
14.27 │  │     │  └─ python >=3.9,<3.10.0a0 , which can be installed;
14.27 │  │     ├─ conda [4.10.0|4.10.1|...|4.9.2] would require
14.27 │  │     │  └─ python >=3.6,<3.7.0a0 , which can be installed;
14.27 │  │     ├─ conda [4.10.0|4.10.1|...|4.9.2] would require
14.27 │  │     │  └─ python_abi 3.6 *_pypy36_pp73, which can be installed (as previously explained);
14.27 │  │     ├─ conda [4.10.1|4.10.2|...|4.13.0] would require
14.27 │  │     │  └─ python_abi 3.7 *_pypy37_pp73, which requires
14.27 │  │     │     └─ python 3.7.* *_73_pypy, which can be installed;
14.27 │  │     └─ conda [4.6.1|4.6.10|...|4.8.3] would require
14.27 │  │        └─ python >=2.7,<2.8.0a0 , which can be installed;
14.27 │  ├─ constructor [3.10.0|3.11.0|...|3.9.3] would require
14.27 │  │  └─ conda >=4.6  with the potential options
14.27 │  │     ├─ conda [22.11.1|22.9.0|...|4.14.0], which can be installed (as previously explained);
14.27 │  │     ├─ conda [22.11.1|22.9.0|...|24.9.2], which can be installed (as previously explained);
14.27 │  │     ├─ conda [22.9.0|4.10.0|...|4.9.2], which can be installed (as previously explained);
14.27 │  │     ├─ conda [22.11.1|22.9.0|4.14.0], which can be installed (as previously explained);
14.27 │  │     ├─ conda [22.11.1|22.9.0|...|4.9.2], which can be installed (as previously explained);
14.27 │  │     ├─ conda [22.11.1|22.9.0|...|4.14.0], which can be installed (as previously explained);
14.27 │  │     ├─ conda [22.11.1|22.9.0|...|4.9.2], which can be installed (as previously explained);
14.27 │  │     ├─ conda [4.10.0|4.10.1|...|4.9.2], which can be installed (as previously explained);
14.27 │  │     ├─ conda [4.10.0|4.10.1|...|4.9.2], which can be installed (as previously explained);
14.27 │  │     ├─ conda [4.10.1|4.10.2|...|4.13.0], which can be installed (as previously explained);
14.27 │  │     ├─ conda [4.6.1|4.6.10|...|4.8.3], which can be installed (as previously explained);
14.27 │  │     ├─ conda [23.1.0|23.3.0|...|23.7.3] would require
14.27 │  │     │  └─ pypy3.8 >=7.3.11  with the potential options
14.27 │  │     │     ├─ pypy3.8 7.3.11 would require
14.27 │  │     │     │  └─ python 3.8.* *_73_pypy, which can be installed;
14.27 │  │     │     └─ pypy3.8 7.3.11 would require
14.27 │  │     │        └─ python 3.8.* *_7.3_pypy, which can be installed;
14.27 │  │     ├─ conda [23.1.0|23.3.0|...|23.7.4] would require
14.27 │  │     │  └─ pypy3.9 [>=7.3.11 |>=7.3.12 ] with the potential options
14.27 │  │     │     ├─ pypy3.9 [7.3.11|7.3.12|7.3.13|7.3.15|7.3.16], which can be installed (as previously explained);
14.27 │  │     │     └─ pypy3.9 7.3.12 would require
14.27 │  │     │        └─ python 3.9.* *_7.3_pypy, which can be installed;
14.27 │  │     └─ conda [23.10.0|23.11.0|...|24.9.2] would require
14.27 │  │        └─ python_abi 3.12.* *_cp312, which requires
14.27 │  │           └─ python 3.12.* *_cpython, which can be installed;
14.27 │  └─ constructor [3.10.0|3.11.0|3.9.3] would require
14.27 │     └─ __osx, which is missing on the system;
14.27 └─ python 3.13**  is not installable because there are no viable options
14.27    ├─ python [3.13.0|3.13.1] conflicts with any installable versions previously reported;
14.27    └─ python [3.13.0rc1|3.13.0rc2|3.13.0rc3] would require
14.27       └─ _python_rc, which does not exist (perhaps a missing channel).
------
```

Ok to settle for 3.12?

Reply:
Oh right, last I checked conda hasn't been packaged for 3.13 yet :/

Let's do 3.12.